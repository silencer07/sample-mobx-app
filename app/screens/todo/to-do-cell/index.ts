export * from "./to-do-cell"
