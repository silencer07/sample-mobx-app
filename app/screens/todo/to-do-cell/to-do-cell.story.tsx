import * as React from 'react'
import { storiesOf } from '@storybook/react-native'
import { StoryScreen, Story, UseCase } from '../../../../storybook/views'
import { ToDoCell } from './'
import { ToDoModel } from '../../../models/to-do/to-do'
import { Wallpaper } from '../../../components/wallpaper'
import { clone } from 'ramda'

const todo = {
  id: 1,
  title: 'Do all the things!',
  isComplete: false
}
export const testTodo = ToDoModel.create(todo)

storiesOf('ToDoCell', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Style Presets', () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <Wallpaper/>
        <ToDoCell todo={testTodo} onToggle={testTodo.toggleIsComplete}/>
      </UseCase>
      <UseCase text="Secondary" usage="Centered">
        <Wallpaper/>
        <ToDoCell preset="secondary" todo={testTodo} onToggle={testTodo.toggleIsComplete}/>
      </UseCase>
    </Story>
  ))
  .add('Interaction with ToDo prop', () => {
    const copy = clone(testTodo)
    copy.isComplete = true

    const copy2 = clone(testTodo)
    copy2.title = null

    return (
      <Story>
        <UseCase text="Completed ToDo" usage="When todo is completed by default">
          <Wallpaper/>
          <ToDoCell todo={copy} onToggle={copy.toggleIsComplete}/>
        </UseCase>
        <UseCase text="Title is null" usage="Shows a default title when nothing was supplied">
          <Wallpaper/>
          <ToDoCell todo={copy2} onToggle={copy.toggleIsComplete}/>
        </UseCase>
      </Story>
    )
  })
