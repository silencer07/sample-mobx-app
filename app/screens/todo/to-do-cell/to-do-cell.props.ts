import { ViewStyle, ViewProperties } from "react-native"
import { ToDoCellPresetNames } from "./to-do-cell.presets"
import { ToDo } from "../../../models/to-do/to-do"

export interface ToDoCellProps extends ViewProperties {
  /**
   * The ToDo being rendered
   */
  todo: ToDo

  /**
   * An optional style override useful for padding & margin.
   */
  style?: ViewStyle

  /**
   * One of the different types of text presets.
   */
  preset?: ToDoCellPresetNames

  /**
   * What happens when you toggle the checkbox
   */
  onToggle?(): void
}
