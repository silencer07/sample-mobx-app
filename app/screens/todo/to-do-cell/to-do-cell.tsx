import * as React from 'react'
import { View } from 'react-native'
import { Text } from '../../../components/text'
import { viewPresets } from './to-do-cell.presets'
import { ToDoCellProps } from './to-do-cell.props'
import { Checkbox } from '../../../components/checkbox'
import { observer } from 'mobx-react'

/**
 * Stateless functional component for your needs
 *
 * Component description here for TypeScript tips.
 */
export const ToDoCell: React.FunctionComponent<ToDoCellProps> = observer(({ preset = 'primary', style: styleOverride, todo, onToggle, ...rest }) => {
  const viewPresetToUse = viewPresets[preset] || viewPresets.primary
  const viewStyle = { ...viewPresetToUse, ...styleOverride }

  return (
    <View style={viewStyle} {...rest} >
      <Checkbox onToggle={onToggle} value={todo.isComplete} />
      <Text text={todo.title || 'Untitled'} />
    </View>
  )
})
