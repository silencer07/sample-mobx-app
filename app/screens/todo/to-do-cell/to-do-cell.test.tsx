import { ToDoCell } from './to-do-cell'
import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { testTodo } from './to-do-cell.story'
import { clone } from 'ramda';

describe('<ToDoCell />', () => {
  test('primary style preset', async () => {
    const renderer = ShallowRenderer.createRenderer()
    const component = renderer.render(<ToDoCell todo={testTodo} />)
    expect(component).toMatchSnapshot()
  })

  test('secondary style preset', async () => {
    const renderer = ShallowRenderer.createRenderer()
    const component = renderer.render(<ToDoCell preset="secondary" todo={testTodo} />)
    expect(component).toMatchSnapshot()
  })

  test('Toggle todo complete', async () => {
    const copy = clone(testTodo)
    copy.isComplete = true

    const renderer = ShallowRenderer.createRenderer()
    const component = renderer.render(<ToDoCell todo={copy} />)
    expect(component).toMatchSnapshot()
  })

  test('Title is null', async () => {
    const copy = clone(testTodo)
    copy.title = null

    const renderer = ShallowRenderer.createRenderer()
    const component = renderer.render(<ToDoCell todo={copy} />)
    expect(component).toMatchSnapshot()
  })
})
