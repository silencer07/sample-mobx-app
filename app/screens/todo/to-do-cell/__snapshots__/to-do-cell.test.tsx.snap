// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoCell /> Title is null 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "flexDirection": "row",
      "paddingHorizontal": 8,
      "paddingVertical": 8,
    }
  }
>
  <Checkbox
    value={false}
  />
  <Text
    text="Untitled"
  />
</View>
`;

exports[`<ToDoCell /> Toggle todo complete 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "flexDirection": "row",
      "paddingHorizontal": 8,
      "paddingVertical": 8,
    }
  }
>
  <Checkbox
    value={true}
  />
  <Text
    text="Do all the things!"
  />
</View>
`;

exports[`<ToDoCell /> primary style preset 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "flexDirection": "row",
      "paddingHorizontal": 8,
      "paddingVertical": 8,
    }
  }
>
  <Checkbox
    value={false}
  />
  <Text
    text="Do all the things!"
  />
</View>
`;

exports[`<ToDoCell /> secondary style preset 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 8,
      "paddingVertical": 8,
    }
  }
>
  <Checkbox
    value={false}
  />
  <Text
    text="Do all the things!"
  />
</View>
`;
